hnn_core/network.py

```diff
@@ -219,47 +286,412 @@ def __repr__(self):
-        # However, they must be present to be included in gid_ranges!
-        origin = self.pos_dict['origin']
-
-        # COMMON FEEDS
-        self.pos_dict['common'] = [origin for i in range(self._n_common_feeds)]
-
-        # UNIQUE FEEDS
-        for key in self._p_unique.keys():
-            # create the pos_dict for all the sources
-            self.pos_dict[key] = [origin for i in range(self.n_cells)]
-
-        # Now add the names of the feeds to a list
-        self.feedname_list.append('common')
-        # grab the keys for the unique set of inputs and sort the names
-        # append them to the src list along with the number of cells
-        unique_keys = sorted(self._p_unique.keys())
-        self.feedname_list += unique_keys
+        if T < 0.:
+            raise ValueError('End time of Poisson input cannot be negative')
+        tstop = self.cell_response.times[-1]
+        if T > tstop:
+            raise ValueError(f'End time of Poisson drive cannot exceed '
+                             f'simulation end time {tstop}. Got {T}.')
+        duration = T - t0
+        if duration < 0.:
+            raise ValueError('Duration of Poisson drive cannot be negative')
+
+        if not isinstance(rate_constants, dict):
+            raise ValueError('rate_constants must be a dict of floats')
+        if not set(rate_constants.keys()).issubset(self.cellname_list):
+            raise ValueError(
+                f"Rate constants should be defined for {self.cellname_list}. "
+                f"Got {list(rate_constants.keys())}")
+
+        drive = NetworkDrive()
+        drive['type'] = 'poisson'
+        drive['cell_specific'] = True
+        drive['seedcore'] = seedcore
+
+        drive['dynamics'] = dict(t0=t0, T=T, rate_constants=rate_constants)
+        drive['events'] = []
+        self._attach_drive(name, drive, weights_ampa, weights_nmda, location,
+                           space_constant, dispersion_time)
+
+    def add_bursty_drive(self, name, distribution, t0, sigma_t0, T, burst_f,
+                         burst_sigma_f, numspikes, repeats, weights_ampa,
+                         weights_nmda, location, dispersion_time=0.1,
+                         space_constant=100., seedcore=42):
+        """Add a bursty (rhythmic) external drive to all cells of the network
 
+        Parameters
+        ----------
+        name : str
+            Unique name for drive
+        distribution : str
+            The distribution for each burst. Either 'normal' or 'uniform'.
+        t0 : float
+            Start time of the burst trains
+        sigma_t0 : float
+            If greater than 0 randomize start times with standard deviation
+            sigma_t0.
+        T : float
+            End of burst trains
+        burst_f : float
+            The frequency of input bursts.
+        burst_sigma_f : float
+            The standard deviation of spike times within burst. Only applied
+            to 'normal' distribution.
```

Comment:
I think this parameter should not be required by default. It's a bit odd to pass this if distribution='uniform'

Reply:
This PR recapitulates all the old 'feeds'--could we leave this for another PR? I don't know of any examples in which a `uniform` drive is used. What's the use case?

Reply:
not sure, we'll have to ask folks from Jones lab. Okay to leave this to another PR. Are you keeping track of those? :)

Reply:
> Are you keeping track of those? :)

I’ll take a moment next to go through review comments so far and open new issues to the ones I propose separating.